python/ray/tune/trial.py

```diff
@@ -592,10 +593,15 @@ def set_experiment_tag(self, experiment_tag):
         self.experiment_tag = experiment_tag
         self.invalidate_json_state()
 
-    def write_error_log(self, error_msg):
+    def write_error_log(self, error_msg, exc: Optional[Exception] = None):
         if error_msg and self.logdir:
             self.num_failures += 1
             self.error_file = os.path.join(self.logdir, "error.txt")
+            if exc and isinstance(exc, RayTaskError):
```

Comment:
should we guard against the case where there is no pickled error file for some reason?

Reply:
yes, this is done at `ResultGrid._populate_exception`. So if there is a pickled_error file, we will load up the original RayTaskError, otherwise, return a TuneError.